client/lib/signup/step-actions/index.js

```diff
@@ -350,10 +351,8 @@ export function setDesignOnSite( callback, { siteSlug, selectedDesign } ) {
 
 	const { theme } = selectedDesign;
 
-	Promise.resolve()
```

Comment:
Why on earth was this necessary 🤔

Reply:
Sometimes folks do this to delay the request to the next event loop tick, to make sure some other `promise.then` callback runs first. Not a great way to write async code, but it happens 🙂